### DISCUSSION TITLE
documentation NewPath

### DISCUSSION THREAD

**Discussion by bytelinker** [pydantic/pydantic#7320] [Pydantic V2]
Hello,

NewPath is documented at https://docs.pydantic.dev/latest/api/types/#pydantic.types.NewPath

Where do I find the documentation to import Annotated, Path, PathType, please?

Thank you very much
bytelinker

**Comment by Viicos** [comment]
Why do you want to import Annotated, Path, PathType? To use `NewPath`, you can use the following import: `from pydantic import NewPath`

**Comment by onecrayon** [comment]
Annotated is part of the standard library:

```python
from typing import Annotated
```

PathType is part of Pydantic's typing package:

```python
from pydantic.typing import PathType
```

And I'm not 100% sure, but I'm guessing that Path is from the standard library, as well:

```python
from pathlib import Path
```

**Comment by bytelinker** [comment]
from pydantic import BaseModel, NewPath
from typing_extensions import Annotated
from pydantic.types import PathType

class Model(BaseModel):
    new_path_a : Annotated[NewPath, PathType('new')] = None
    new_path_b : Annotated[NewPath, PathType('new')] = None

file_a = r'c:\test.txt'
file_b = r'b:\test.txt'

example = Model(new_path_a=file_a, new_path_b=file_b)

But I don't find a file/path type which only evaluates if a theoretical valid path is provided, but ignores, that the file and any component of the path is not accessible.
